daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/DaffodilTDMLDFDLProcessor.scala

```diff
@@ -128,11 +128,12 @@ final class TDMLDFDLProcessorFactory private (
         if (useSerializedProcessor) {
           val os = new java.io.ByteArrayOutputStream()
           val output = Channels.newChannel(os)
-          p.save(output)
+          p.save(output, clearDiagnostics = false) // TDML runner wants to preserve all diagnostics
```

Comment:
As an alternate approach, what are your thoughts on making it so we never clear diagnostics from a DataProcessor, whether it be serialized or not. The benefit is that when you reload a DataProcessor, you still have access to those compile time diagnostics if you want them. So if you want to know if there were any warnings when you compiled some old DataProcessor, you can easily figure that out.

And then to avoid extraneous diagnostics during parse/unparse we change it so DataProcessor diagnostics do not carry over to the ParseResult diagnostics. So there are two completely separate buckets of diagnostics--compile time and runtime. If a user needs access to the compile time diagnostics (like the TDML runner) , it needs to pull them off of the DataProcessor.

In fact, maybe we already do this and the compile-time and run-time diagnosics are already separate? Maybe the fix is to just make it so when we `reload()` we don't output diagnostics from the DataProcessor? In other words, we remove this line:

https://github.com/apache/daffodil/blob/main/daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala#L595

That only affects the CLI. I'm not sure if that is where you were seeing too much diagnostic information.

Reply:
> TDML Runner won't do anything with compilation time diagnostics in its usual path anyway. I noticed that behavior when implementing runtime2's support for returning a processor to TDML Runner.

If at test has <tdml:errors> or <tdml:warnings> then I think TDML runner should look in both compile time and runtime diagnostics, I think...